src/util/stl_common.hpp

```diff
@@ -0,0 +1,58 @@
+/* Copyright © 2017 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef TURI_STL_COMMON_H_
+#define TURI_STL_COMMON_H_
+
+#include <algorithm>
```

Comment:
Combining all the STL headers (and especially adding `using` statements for the types within) doesn't really fit the code style of existing code. I think we should prefer explicit STL includes at each point of use (at least for now, perhaps until we have PCH turned on in the build), and explicit references to `std::`.